### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by AlselwiAisha**
**Project requirements**

- **Design:**

    -  You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
    - Colors.
    - Typography: font face, size and weight.
    -  Layout: composition and space between elements.

- **Interactions:**

- Splash screen
    - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
    - Sign up and log in pages
    - The user should be able to register in the app with full name, email and password (all mandatory).
    - The user can log into the app using email and password.
    - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Home page (categories page)
    - When the user logs in, they are presented with the categories page.
    - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
    - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
    - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page
    - For a given category, the list of transactions is presented, ordered by the most recent.
    - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
    - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page
    - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
    - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new transaction" page
    - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
    - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
    - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- Testing requirements
    - Create unit and integration tests for all the most important components of your RoR application.
- Technical requirements
    - You should use Postgres as your database.
    - You should use devise for authentication.
    - You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
    - You can use a view template engine of your choice (.erb, .slim, .haml).
    - The project should be deployed and accessible online.

### CODE REVIEW

File: app/views/transacts/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<div>
+ <% @header = "NEW TRANSACTION" %>
+  <%= form_with model: @transact, url: category_transacts_path(@category) do |form| %>
+          <div class='main-field'>
+            <%= form.text_field :name,  placeholder: "Name", required: true, class:'text-field'%>
+            <%= form.number_field :amount, placeholder: "Amount (USD)", required: true, class:'text-field' %>
+          </div>
+    <%= form.submit "Save", class:'save-btn'%>
+  <% end %>
+</div>
```
Comment (Abenezer-Tilahun) [inline app/views/transacts/new.html.erb]: - [ ] As stated in the project requirement when creating a new transaction with, indicating their name and amount and categories(Kindly make sure that all are mandatory ).
 - Your new transaction  Form miss the categories as seen in the below screenshot, as stated in the project requirement there should be categories (mandatory at least one)  
 
image

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @AlselwiAisha  

Great job for implemented  the some of the Project Requirement  , but there are some changes that need to be implemented in order to proceed to the next project 👍

## Highlights 👏🏻

🎲 Properly added required models, controllers, and corresponding views
🎲 Devise correctly implemented for authentication!
🎲 Schema follow given ERD!
🎲 Good UI, closely matches given template design!
🎲 App is running as intended!
🎲 ReadMe looks great!
🎲 Good Job for Added required spec test cases and  all test cases are passing! 

image

almost

## Required Changes ♻️
_Check the comments below the review._

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/transacts/new.html.erb (hunk 1 above)
Comment (AlselwiAisha) [reply app/views/transacts/new.html.erb:10]: I am sorry, I don't get the issue

**Review by V-Blaze — APPROVED** [review]
Hi @AlselwiAisha  , 

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

## Highlights

- Fixed issues with Categories💯
-  App works as expected👍
-  User can create account and login🟢 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**